src/asm/output.cpp

```diff
@@ -347,10 +351,57 @@ void out_WriteObject() {
 		writeassert(assert, file);
 }
 
+void out_WriteState() {
+	if (stateFileName.empty())
+		return;
+
+	FILE *file;
+	if (stateFileName != "-") {
+		file = fopen(stateFileName.c_str(), "w");
+	} else {
+		stateFileName = "<stdout>";
+		file = fdopen(STDOUT_FILENO, "w");
+	}
+	if (!file)
+		err("Failed to open state file '%s'", stateFileName.c_str());
+	Defer closeFile{[&] { fclose(file); }};
+
+	// Only write the non-builtin numeric constants
```

Comment:
I think it would be more useful to write more user-defined symbols, such as `EQUS`. To that effect, employing RGBASM's syntax sounds like a good idea (reusing the provided syntax highlighting), and we could use a `.dump.asm` extension in the shell completions.

Perhaps we can add an extra CLI flag that lets the user decide what to request in the dump file; I'd consider using the same syntax as rgbobj for that.